Source/MaestroDebug.cpp

```diff
@@ -81,6 +83,31 @@ void Maestro::PrintMF(const Vector<MultiFab>& MF) {
     }
 }
 
+
+void Maestro::PrintBA(const Vector<BoxArray>& ba) {
+    // timer for profiling
+    BL_PROFILE_VAR("Maestro::PrintBA()", PrintBA);
+
+    for (int lev = 0; lev <= finest_level; ++lev) {
+    
+      for (int i = 0; i < ba[lev].size(); ++i) {
+	  Print() << "Grid #" << i << std::endl;
+	
+	  const Box& validBox = ba[lev][i];
+	  auto lo = validBox.loVect3d();
+	  auto hi = validBox.hiVect3d();
+
+	  std::cout << "Level " << lev << std::endl;
+	  std::cout << "valid box ";
+	  for (auto n = 0; n < AMREX_SPACEDIM; ++n) {
```

Comment:
Be careful mixing Print() and std::cout throughout this routine.
Maybe this whole routine should be in an IOProcessor if test and then just use cout?
Each processor has a copy of the boxarray.